cfme/common/provider.py

```diff
@@ -553,30 +554,22 @@ def process_credential_yaml_key(cls, cred_yaml_key, cred_type=None):
             return cls.get_credentials_from_config(cred_yaml_key, cred_type=cred_type)
 
     # Move to collection
-    @staticmethod
-    def clear_providers_by_class(prov_class, validate=True):
-        """ Removes all providers that are an instance of given class or one of it's subclasses
-        """
-        from utils.providers import ProviderFilter, list_providers
-        pf = ProviderFilter(classes=[prov_class])
-        provs = list_providers(filters=[pf], use_global_filters=False)
-
-        # First, delete all
-        deleted_provs = []
-        for prov in provs:
-            existed = prov.delete_if_exists(cancel=False)
-            if existed:
-                deleted_provs.append(prov)
-        # Then, check that all were deleted
-        if validate:
-            for prov in deleted_provs:
-                prov.wait_for_delete()
-
-    # Move to collection
-    @staticmethod
-    def clear_providers():
-        """ Clear all providers on an appliance using UI """
-        BaseProvider.clear_providers_by_class(BaseProvider)
+    @classmethod
+    def clear_providers(cls):
+        """ Clear all providers of given class on the appliance """
+        from utils.appliance import current_appliance as app
```

Comment:
I don't have a better way around this as is. If you can think of something, I'm definitely all ears.